### PR TITLE
Add methods from ACS General Handbook Ch 8

### PR DESCRIPTION
**Pull request by nkrishnaswami**
Add methods for estimating MOE for 
* proportions, 
* ratios, and 
* products

### CODE REVIEW

**Comment by palewire** [comment]
Thank you. I think we will ultimately accept all of these additions to the library. I just want to brush up my own understanding of the math, expand the documentation a little and strength the unittests.

I've pulled down your PR to a branch and pushed some incremental updates to the repo. They don't show up here in the PR but maybe we can use this space to walk through them together. 

The proportion documentation in the ACS handbook uses the single female population in suburban Virginia [as its example](https://www.documentcloud.org/documents/6177941-Acs-General-Handbook-2018-ch08.html#document/p6). I'd like to add those values to your unittest. I've done so [here](https://github.com/datadesk/census-data-aggregator/blob/nkrishnaswami-master/test.py#L101-L107), but I suspect I'm misunderstanding what the expected result should be. What am I missing?

File: census_data_aggregator/__init__.py [hunk 1]
```diff
@@ -214,3 +214,108 @@ def approximate_median(range_list, design_factor=None):
...
+        >>> approximate_proportion((379, 1), (384, 1))
+        (0.9869791666666666, 0.008208247339752435)
+
+    """
+    # Pull out the values
+    numerator_estimate, numerator_moe = numerator_pair
+    denominator_estimate, denominator_moe = denominator_pair
+
+    # Approximate the proportion
+    proportion_estimate = 1.0 * numerator_estimate / denominator_estimate
```
Comment (palewire) [inline census_data_aggregator/__init__.py]: What is the reason for multiplying by 1.0 here? Doesn't the result come out the same regardless?

**Comment by palewire** [comment]
I'm also curious about the reason for multiplying by 1.0 [here](https://github.com/datadesk/census-data-aggregator/blob/nkrishnaswami-master/census_data_aggregator/__init__.py#L253-L254). Doesn't the result come out the same regardless?

File: census_data_aggregator/__init__.py (hunk 1 above)
Comment (nkrishnaswami) [reply census_data_aggregator/__init__.py]: It's omissible in Python 3 , but in Python 2, if division arguments are given as integers, it defaults to integer division. Happy to remove if it you prefer.

**Comment by nkrishnaswami** [comment]
There is a missing squaring of a denominator for ratio and proportions. I've updated that, and the testcases to check the examples in the Handbook.

**Comment by palewire** [comment]
Thanks for the updates.

How would you feel about adding `from __future__ import division` to the top of the script to provide reverse compatibility on division of integers?

File: census_data_aggregator/__init__.py (hunk 1 above)
Comment (palewire) [reply census_data_aggregator/__init__.py]: How would you feel about substituting from __future__ import division to the top of the script to provide reverse compatibility on division of integers?

Comment (nkrishnaswami) [reply census_data_aggregator/__init__.py]: Forgot about that one!  Done.

**Comment by palewire** [comment]
Screenshot from 2019-07-01 08-22-22

As far as the proportion formula, am I correct that a synonymous implementation would be something like:

```
    squared_proportion_moe = numerator_moe**2 - (proportion_estimate**2 * denominator_moe**2)
    proportion_moe = (1.0 / denominator_estimate) * math.sqrt(squared_proportion_moe)
```

**Comment by palewire** [comment]
Educate me. What is the functional difference between the proportion and the ratio code?

**Comment by palewire** [comment]
I've also noticed a percent change formula in the handbook. You think we should considering adding that too, or not?

**Comment by nkrishnaswami** [comment]
Whoops, missed the percent change section. I'd be happy to make another PR for that.

**Comment by nkrishnaswami** [comment]
Re proportion vs ratio, the first order Taylor approximation for variance of a function independent random variables X and Y (so we can ignore the covariance term) is
variance approximation for f(X,Y)
I haven't chugged through the algebra, but I expect the sign difference between ratio and proportion comes from whether f(X,Y) is
ratio var approx
or
proportion var approx

**Pull request merged**